internal/statusupdater/statusupdater.go

```diff
@@ -1,24 +1,29 @@
-package module
+package statusupdater
 
 import (
 	"context"
+	"time"
 
 	ootov1alpha1 "github.com/qbarrand/oot-operator/api/v1alpha1"
 	"github.com/qbarrand/oot-operator/internal/daemonset"
 	"github.com/qbarrand/oot-operator/internal/metrics"
 	appsv1 "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-//go:generate mockgen -source=statusupdater.go -package=module -destination=mock_statusupdater.go
+//go:generate mockgen -source=statusupdater.go -package=statusupdater -destination=mock_statusupdater.go
 
 type StatusUpdater interface {
-	UpdateModuleStatus(ctx context.Context,
-		mod *ootov1alpha1.Module,
-		kernelMappingNodes []v1.Node,
-		targetedNodes []v1.Node,
-		dsByKernelVersion map[string]*appsv1.DaemonSet) error
+	ModuleUpdateStatus(ctx context.Context, mod *ootov1alpha1.Module, kernelMappingNodes []v1.Node,
+		targetedNodes []v1.Node, dsByKernelVersion map[string]*appsv1.DaemonSet) error
+	PreflightPresetVerificationStatus(ctx context.Context, pv *ootov1alpha1.PreflightValidation,
+		status *ootov1alpha1.CRStatus) error
+	PreflightSetVerificationStatus(ctx context.Context, preflight *ootov1alpha1.PreflightValidation, status *ootov1alpha1.CRStatus,
+		verificationStatus string, message string) error
+	PreflightSetVerificationStage(ctx context.Context, preflight *ootov1alpha1.PreflightValidation,
+		status *ootov1alpha1.CRStatus, stage string) error
```

Comment:
Should we have methods updating `Module` and `Preflight` in two distinct interfaces? The `Module` reconciler does not need to update `Preflight` statuses and vice versa.

Reply:
We can split it into 2 public interfaces, but then it means also initializing to 2 interfaces objects in the main and passing them accordingly to preflight package and to module reconciuler. Keep in mind that in that case we will probably need to define ( in the future) a third inteface, for hub and spoke.  I prefer to keep them all in single interfaces, but i won't object to strongly to splitting them into multiple public interfaces

Reply:
I don't mind initializing several objects in `main()` and I think it is generally more convenient to have smaller interfaces.

Reply:
fixed